### ISSUE TITLE
Error setting custom MX record for a domain.

### ISSUE THREAD

**Issue by jvolkenant** [mail-in-a-box/mailinabox#1781]
I have multiple domains on my mail in a box install.  I wanted to test mta-sts when a custom MX record is set to something that is not the mail in a box install itself to see how the system will handle it. However I get this error when setting a custom MX record for a domain.

```
Jun 15 15:48:09 m start[11988]: Zone not read, error: Syntax error, could not parse the RR's rdata at /etc/nsd/zones/example.com.txt line 13
Jun 15 15:48:09 m start[11988]: [2020-06-15 15:48:09,369] ERROR in app: Exception on /dns/custom/example.com/MX [POST]
Jun 15 15:48:09 m start[11988]: Traceback (most recent call last):
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/app.py", line 2447, in wsgi_app
Jun 15 15:48:09 m start[11988]:     response = self.full_dispatch_request()
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/app.py", line 1952, in full_dispatch_request
Jun 15 15:48:09 m start[11988]:     rv = self.handle_user_exception(e)
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/app.py", line 1821, in handle_user_exception
Jun 15 15:48:09 m start[11988]:     reraise(exc_type, exc_value, tb)
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
Jun 15 15:48:09 m start[11988]:     raise value
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
Jun 15 15:48:09 m start[11988]:     rv = self.dispatch_request()
Jun 15 15:48:09 m start[11988]:   File "/usr/local/lib/mailinabox/env/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
Jun 15 15:48:09 m start[11988]:     return self.view_functions[rule.endpoint](**req.view_args)
Jun 15 15:48:09 m start[11988]:   File "/root/mailinabox/management/daemon.py", line 53, in newview
Jun 15 15:48:09 m start[11988]:     return viewfunc(*args, **kwargs)
Jun 15 15:48:09 m start[11988]:   File "/root/mailinabox/management/daemon.py", line 315, in dns_set_record
Jun 15 15:48:09 m start[11988]:     return do_dns_update(env) or "Something isn't right."
Jun 15 15:48:09 m start[11988]:   File "/root/mailinabox/management/dns_update.py", line 87, in do_dns_update
Jun 15 15:48:09 m start[11988]:     sign_zone(domain, zonefile, env)
Jun 15 15:48:09 m start[11988]:   File "/root/mailinabox/management/dns_update.py", line 668, in sign_zone
Jun 15 15:48:09 m start[11988]:     dnssec_keys["ZSK"],
Jun 15 15:48:09 m start[11988]:   File "/root/mailinabox/management/utils.py", line 123, in shell
Jun 15 15:48:09 m start[11988]:     ret = getattr(subprocess, method)(cmd_args, **kwargs)
Jun 15 15:48:09 m start[11988]:   File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
Jun 15 15:48:09 m start[11988]:     raise CalledProcessError(retcode, cmd)
Jun 15 15:48:09 m start[11988]: subprocess.CalledProcessError: Command '['/usr/bin/ldns-signzone', '-e', '20200715', '-n', '/etc/nsd/zones/example.com.txt', '/tmp/Kexample.com.+007+46974', '/tmp/Kfireislandak.com.+007+32447']' returned non-zero exit status 1.

```

I end up with two MX records in the zone file, the one for the mail in a box server, and one for the custom record inputted via the admin page
```
root@m:/etc/nsd/zones# grep MX /etc/nsd/zones/example.com.txt |wc -l
2
```

It would make me think that you could set a custom MX record since it is available in the drop down for custom DNS.

Does mail in a box support setting the MX record to something not on this box? (i.e. using the box  as a nameserver and webserver only)

If you are able to set a MX record that is not the MIAB server, the settings for mta-sts should not populate for that domain since the destination server may not support it. (If MX record is being being manually changed, the mta-sts settings should be manually changed too)

**Comment by JoshData** [comment]
Setting MX records might be more useful for subdomains that are not managed by the box.

If we want to support web and DNS without mail for a domain, we should modify the source code to make that cleanly possible. I think we just need a way to add a domain to the set returned by get_web_domains in management/web_update.py.

I don't know the specific cause for the error you are seeing. It has to do with DNSSEC and not with MX records specifically.

**Comment by jvolkenant** [comment]
It appears the problem is entirely my fault. I did not follow the proper formatting for the record. Also I could not point the mx record to a CNAME.  I had to create a new A record for the destination and follow the format for MX record "10 mx.somemailserver.com."

I'll create a new issue about the original problem I expected would exist.  If MIAB is the nameserver for a domain, but not the MX server, mta-sts dns and nginx config is still generated.  Causing the MX record to point to one place, and the ".well-known/mta-sts.txt" file to still point to the MIAB domain.
